addons/website/models/mixins.py

```diff
@@ -367,7 +367,7 @@ def _search_fetch(self, search_detail, search, limit, order):
             limit=limit,
             order=search_detail.get('order', order)
         )
-        count = model.search_count(domain)
+        count = model.search_count(domain) if limit else len(results)
```

Comment:
What about `if limit and limit == len(results)` ?
(since if we have 3 results out of a max of 5... we already have the result too, right ?)